Data/ByteString.hs

```diff
@@ -1147,6 +1148,19 @@ findIndex k (PS x s l) = accursedUnutterablePerformIO $ withForeignPtr x $ \f ->
                                   else go (ptr `plusPtr` 1) (n+1)
 {-# INLINE findIndex #-}
 
+-- | The 'findIndexEnd' function takes a predicate and a 'ByteString' and
+-- returns the index of the last element in the ByteString
+-- satisfying the predicate.
+findIndexEnd :: (Word8 -> Bool) -> ByteString -> Maybe Int
+findIndexEnd k (PS x s l) = accursedUnutterablePerformIO $ withForeignPtr x $ \ f -> go (f `plusPtr` s) (l-1)
+  where
+    go !ptr !n | n < 0     = return Nothing
+               | otherwise = do w <- peekByteOff ptr n
+                                if k w
+                                  then return (Just n)
+                                  else go ptr (n-1)
+{-# INLINE findIndexEnd #-}
```

Comment:
This is very similar to `elemIndexEnd`, although some style and variable name differences make that a bit hard to see.

https://github.com/haskell/bytestring/blob/95fe6bdf13c9cc86c1c880164f7844d61d989574/Data/ByteString.hs#L1092-L1109